src/auth/src/credentials/external_account.rs

```diff
@@ -241,6 +267,16 @@ impl Builder {
         self
     }
 
+    /// bring your own custom implementation of
+    /// SubjectTokenProvider for OIDC/SAML credentials.
+    pub fn with_subject_token_provider<T: SubjectTokenProvider + 'static>(
+        mut self,
+        subject_token_provider: T,
+    ) -> Self {
+        self.subject_token_provider = Some(Box::new(subject_token_provider));
+        self
+    }
```

Comment:
If we are taking subject token provider as an input in the builder, we need to make some more changes as well. It becomes tricky now.  There is also a credential source in the config they initialized the builder with.

Reply:
when setting the subject token provider via the `Bulder`, it overrides the credential source from the config

Reply:
Hmm that maybe ok. When implementing this change lets make sure to document that clearly.

Reply:
Another way could be to introduce `new(subject_token_provider)`. But that means we add builder methods to provide all other fields like impersonation url, provider ppol, token url etc.